src/ServiceControl.AcceptanceTests/AcceptanceTest.cs

```
@@ -419,113 +288,358 @@ protected byte[] DownloadData(string url)
+                var cts = new CancellationTokenSource();
+                app.Properties[AppBuilderExtensions.HostOnAppDisposing] = cts.Token;
+                bootstrapper.WebApp = new Disposable(() => cts.Cancel(false));
+                bootstrapper.Startup.Configuration(app);
+                var appFunc = app.Build();
 
-            return response.StatusCode;
+                Handler = new OwinHttpMessageHandler(appFunc)
+                {
+                    UseCookies = false,
+                    AllowAutoRedirect = false
+                };
+                httpClient = new HttpClient(Handler);
+            }
         }
 
-        public IScenarioWithEndpointBehavior<T> Define<T>() where T : ScenarioContext, new()
+        private class Disposable : MarshalByRefObject, IDisposable
         {
-            Func<T> instance = () => new T();
-            return Define(instance);
+            private readonly Action dispose;
+
+            public Disposable(Action dispose)
+            {
+                this.dispose = dispose;
+            }
+
+            public void Dispose()
+            {
+                dispose();
+            }
         }
 
-        public IScenarioWithEndpointBehavior<T> Define<T>(T context) where T : ScenarioContext, new()
+        private class MessageMapperInterceptor : Feature
         {
-            return Define(() => context);
+            public MessageMapperInterceptor()
+            {
+                DependsOn<JsonSerialization>();
+            }
+
+            protected override void Setup(FeatureConfigurationContext context)
+            {
+                context.Container.ConfigureComponent<IMessageMapper>(builder => new MessageMapperWrapper(), DependencyLifecycle.SingleInstance);
+            }
         }
 
-        public IScenarioWithEndpointBehavior<T> Define<T>(Func<T> contextFactory) where T : ScenarioContext, new()
+        private class MessageMapperWrapper : IMessageMapper
```

If we were to change the default message mapper in ServiceControl this would hide that change. I don't know that there's anything we can do about that :(